twitter_tunes/tests/test_bot.py

```diff
@@ -0,0 +1,227 @@
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.get_redis_data')
+@mock.patch('twitter_tunes.scripts.twitter_api')
+@mock.patch('twitter_tunes.scripts.youtube_api')
+@mock.patch('twitter_tunes.scripts.twitter_bot.tweepy.API.trends_place')
+def test_main_bad_get_trends(trends_place, yt, tw, get, set):
+    """Test if main does stuff if tweepy goes horribly wrong.
+
+    Make sure it can keep going."""
+    from tweepy import RateLimitError
+    get.return_value = {}
+    trends_place.side_effect = RateLimitError("Slow Down!")
+    assert twitter_bot.main() == u'Something went horribly wrong.'
+
+
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.set_redis_data')
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.get_redis_data')
+@mock.patch('twitter_tunes.scripts.twitter_api')
+@mock.patch('tweepy.API')
+@mock.patch('twitter_tunes.scripts.twitter_bot.youtube_api.youtube_search')
+def test_main_bad_youtube(youtube_search, tweepy, ta, get, set):
+    """Test if main does stuff if twitter goes horribly wrong.
+
+    Make sure it can keep going."""
+    get.side_effect = redis_side_effect
+    from apiclient.errors import HttpError
+    youtube_search.side_effect = HttpError('Uhh', b'youtube broke.')
+    assert twitter_bot.main() == u'Something went horribly wrong.'
+
+
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.get_redis_data')
+def test_main_bad_redis(get_redis_data):
+    """Test if main does stuff if redis goes horribly wrong."""
+    from requests.exceptions import ConnectionError
+    get_redis_data.side_effect = ConnectionError
+    assert twitter_bot.main() == u'Something went horribly wrong.'
+
+
+def test_bot_create_message_known_params():
+    """Test to see bot can return a message.
+
+    Should contain the trend name and a youtube url.
+    """
+    url = u'https://www.youtube.com/watch?v=rTfa-9aCTYg'
+    message = twitter_bot.create_message(u'A Trend', url)
+    assert u'A Trend' in message and url in message
+
+
+@mock.patch('twitter_tunes.scripts.youtube_api.get_link')
+def test_bot_message_function_params(get_link):
+    """Test to see bot can return a message.
+
+    Message should be based on the returns of other functions.
+    """
+    from twitter_tunes.scripts import parser
+    trend = u"#StoryFromNorthAmerica"
+    parse_trend = parser.parse_trend(trend)
+    # Results that would come from searching this trend.
+    # Saved locally to prevent api calls each test.
+    get_link.return_value = (u'https://www.youtube.com/watch?v=ms2klX-puUU',
+                             True)
+    url = get_link(parse_trend)
+    message = twitter_bot.create_message(trend, url[0])
+    assert (u'#StoryFromNorthAmerica' in message and
+            u'https://www.youtube.com/watch?v=ms2klX-puUU' in message)
+
+
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.set_redis_data')
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.get_redis_data')
```

Comment:
It occurs to me that mocking redis itself rather than your wrapper functions around redis might be more instructive.